pandas/tests/test_frame.py

```diff
@@ -8881,6 +8881,16 @@ def test_drop(self):
         df.drop(labels=df[df.b>0].index, inplace=True)
         assert_frame_equal(df,expected)
 
+        # multi-index issue
+        # origin of GH 11640
+        df = pd.DataFrame(columns=['a', 'b', 'c', 'd'],
+                          data=[[1, 'b1', 'c1', 3],
+                                [1, 'b2', 'c2', 4]])
+        dg = df.pivot_table(index='a', columns=['b', 'c'], values='d')
+        dg = dg.reset_index()
```

Comment:
So here's how to construct the frames (df1) that you are using. Indexing using ``[]`` 'work's when you only pass in a single level, but this can easily be ambiguous. see df2, which is a much more general case. This should raise if it cannot index properly (e.g. the ``level`` is not provided), not magically work, this just makes the API MORE confusing. ``a`` and ``df['a']`` **happen** to be the same, IN THIS CASE, but not always.

```
In [39]: df1 = DataFrame([[1,3,4]],columns=pd.MultiIndex.from_tuples([('a',''),('b1','c1'),('b2','c2')],names=['b','c']))

In [40]: df1['a']
Out[40]: 
0    1
Name: a, dtype: int64

In [41]: df2 = DataFrame([[1,2,3,4]],columns=pd.MultiIndex.from_tuples([('a','a1'),('a','a2'),('b1','c1'),('b2','c2')],names=['b','c']))     

In [42]: df2['a']
Out[42]: 
c  a1  a2
0   1   2
```